src/Types.jl

```diff
@@ -88,18 +88,46 @@ Base.:(==)(r1::GitRepo, r2::GitRepo) =
     r1.source == r2.source && r1.rev == r2.rev && r1.subdir == r2.subdir
 
 
-Base.@kwdef mutable struct PackageSpec
-    name::Union{Nothing,String} = nothing
-    uuid::Union{Nothing,UUID} = nothing
-    version::Union{Nothing,VersionTypes,String} = VersionSpec()
-    tree_hash::Union{Nothing,SHA1} = nothing
-    repo::GitRepo = GitRepo()
-    path::Union{Nothing,String} = nothing
-    pinned::Bool = false
+mutable struct PackageSpec
+    name::Union{Nothing,String}
+    uuid::Union{Nothing,UUID}
+    version::Union{Nothing,VersionTypes,String}
+    tree_hash::Union{Nothing,SHA1}
+    repo::GitRepo
+    path::Union{Nothing,String}
+    pinned::Bool
     # used for input only
-    url = nothing
-    rev = nothing
-    subdir = nothing
+    url
+    rev
+    subdir
+    function PackageSpec(; name::Union{Nothing,String} = nothing,
+                           uuid::Union{Nothing,UUID,AbstractString} = nothing,
+                           version::Union{Nothing,VersionTypes,String} = VersionSpec(),
+                           tree_hash::Union{Nothing,SHA1} = nothing,
+                           repo::GitRepo = GitRepo(),
+                           path::Union{Nothing,String} = nothing,
+                           pinned::Bool = false,
+                           url = nothing,
+                           rev = nothing,
+                           subdir = nothing)
+        if uuid isa AbstractString
+            _uuid = Base.UUID(uuid)
+        else
+            _uuid = uuid
+        end
+        return new(
+            name,
+            _uuid,
+            version,
+            tree_hash,
+            repo,
+            path,
+            pinned,
+            url,
+            rev,
+            subdir,
+        )
```

Comment:
```suggestion
        return new(name, uuid, version, tree_hash, repo, path, pinned, url, rev, subdir)
```

Reply:
GitHub won't let me "apply" this suggestion directly, but I have made this change manually.